packages/infrastructure/kube_airbyte/main.tf

```diff
@@ -0,0 +1,1301 @@
+  az_spread_preferred                  = var.sla_target >= 2
+  spot_nodes_enabled                   = var.spot_nodes_enabled
+  burstable_nodes_enabled              = var.burstable_nodes_enabled
+  controller_nodes_enabled             = var.controller_nodes_enabled
+  extra_labels                         = data.pf_kube_labels.labels.labels
+}
+
+module "namespace" {
+  source = "../kube_namespace"
+
+  namespace = var.namespace
+}
+
+/***************************************
+* Database Backend
+***************************************/
+
+module "database" {
+  source = "../kube_pg_cluster"
+
+  pg_version                           = "14.17" # based on minimum supported version https://docs.temporal.io/clusters#dependency-versions
+  pg_cluster_namespace                 = local.namespace
+  pg_initial_storage_gb                = var.pg_initial_storage_gb
+  pg_instances                         = var.sla_target >= 2 ? 2 : 1
+  pg_smart_shutdown_timeout            = 1
+  pg_minimum_memory_mb                 = 500
+  aws_iam_ip_allow_list                = var.aws_iam_ip_allow_list
+  pull_through_cache_enabled           = var.pull_through_cache_enabled
+  pgbouncer_pool_mode                  = "session"
+  burstable_nodes_enabled              = var.burstable_nodes_enabled
+  spot_nodes_enabled                   = var.spot_nodes_enabled
+  controller_nodes_enabled             = false
+  monitoring_enabled                   = var.monitoring_enabled
+  panfactum_scheduler_enabled          = var.panfactum_scheduler_enabled
+  instance_type_anti_affinity_required = var.sla_target == 3
+  vpa_enabled                          = var.vpa_enabled
+
+  pg_backup_directory      = var.db_backup_directory
+  pg_recovery_mode_enabled = var.db_recovery_mode_enabled
+  pg_recovery_directory    = var.db_recovery_directory
+  pg_recovery_target_time  = var.db_recovery_target_time
+}
+
+/***************************************
+* Service Account with S3 Permissions
+***************************************/
+
+resource "kubernetes_service_account" "airbyte_sa" {
+  metadata {
+    name      = "airbyte-admin"
+    namespace = local.namespace
+    labels    = data.pf_kube_labels.labels.labels
+  }
+}
+
+/***************************************
+* S3 Storage Configuration
+***************************************/
+
+resource "random_id" "airbyte_bucket_name" {
+  byte_length = 8
+  prefix      = "airbyte-"
+}
+
+module "airbyte_bucket" {
+  source      = "../aws_s3_private_bucket"
+  bucket_name = random_id.airbyte_bucket_name.hex
+  description = "Airbyte storage for logs state and workload output"
+
+  intelligent_transitions_enabled = true
+}
+
+data "aws_iam_policy_document" "airbyte_bucket" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:PutObject",
+      "s3:GetObject",
+      "s3:DeleteObject"
+    ]
+    resources = concat(
+      ["${module.airbyte_bucket.bucket_arn}/*"],
+      [for arn in var.additional_s3_bucket_arns : "${arn}/*"]
```

Comment:
I don't think "additional_s3_bucket_arns" is the right descriptor here. Perhaps "connected_s3_bucket_arns"